### PR TITLE
Let Content-Type be set automatically when request body is FormData or URLSearchParams

### LINKED ISSUE #16: form-data content-type header changes and makes the payload invalid
When I use a normal axios request with a `FormData` payload, I get a header like this:

```
content-type: multipart/form-data; boundary=----WebKitFormBoundaryAtMBWBnzlZT7VwNE`
```

And the payload looks like this:

```
------WebKitFormBoundaryAtMBWBnzlZT7VwNE
Content-Disposition: form-data; name="client_id"

stuff
------WebKitFormBoundaryAtMBWBnzlZT7VwNE
Content-Disposition: form-data; name="grant_type"

password
------WebKitFormBoundaryAtMBWBnzlZT7VwNE
Content-Disposition: form-data; name="username"

manufacturer-moderator
------WebKitFormBoundaryAtMBWBnzlZT7VwNE
Content-Disposition: form-data; name="password"

blablatestpass
------WebKitFormBoundaryAtMBWBnzlZT7VwNE--
```

but when I use this adapter, the header changes to:

```
content-type: application/x-www-form-urlencoded
```

And because now the boundary is missing, the payload has become invalid.

### PR DESCRIPTION
**Pull request by Evertt**
Fixes #16

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -103,6 +103,12 @@ function createRequest(config) {
     };
     if (method !== 'GET' && method !== 'HEAD') {
         options.body = config.data;
+
+        // In these cases the browser will automatically set the correct Content-Type,
+        // but only if that header hasn't been set yet. So that's why we're deleting it.
+        if (options.body instanceof FormData || options.body instanceof URLSearchParams) {
+            headers.delete('Content-Type');
```
Comment (agentlogic) [inline index.js:110]: Tested this and it works just fine (after having had issues with file uploads with v0.3.0). But would it be better to do a case-insensitive match for `Content-Type`?

Hopefully this gets merged soon!

**Comment by mark-beeby** [comment]
Can we merge this please? It's a rather fundamental break in the library as is.

File: index.js (hunk 1 above)
Comment (vespaiach) [reply index.js:110]: Thanks

Comment (vespaiach) [reply index.js:110]: actually, Axios has a util function for checking if submited data is formdata or not?

**Pull request merged**